### PR TITLE
Merge 15.1 code freeze into trunk

### PR DESCRIPTION
**Pull request by spencertransier**
This PR merges the changes from the 15.1 code freeze into `trunk`. Includes: 
 - Version bump (`15.1.0.1` because there was an issue with the initial build and I had to create a new build: p1693007584312879/1693004877.995529-slack-C5ALGJYEP )
 - Updated release notes
 - Frozen app strings for translation
 - Added missing xcconfig env var: https://github.com/woocommerce/woocommerce-ios/pull/10544

### CODE REVIEW

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile:19]: @spencertransier  Did you decide to rename `PUBLIC_VERSION_CONFIG_FILE` to `PUBLIC_CONFIG_FILE` so that it has the same name as the environment variable?

File: WooCommerce/Resources/AppStoreStrings.pot
Comment (mokagio) [inline WooCommerce/Resources/AppStoreStrings.pot]: The various `\n` removals were unexpected. The linter didn't complain so I think they'd be effectively harmless. Still, here's a PR to fix them before merging on `trunk` to avoid noise in GlotPress.

https://github.com/woocommerce/woocommerce-ios/pull/10545

**Review by mokagio — APPROVED** [review]
Approving to unblock but I think we should merge https://github.com/woocommerce/woocommerce-ios/pull/10545 into here first @spencertransier

File: fastlane/Fastfile
Comment (spencertransier) [reply fastlane/Fastfile:19]: @mokagio Yep, while I was re-adding the environment variable, I decided to rename the constant to match what you did in https://github.com/wordpress-mobile/WordPress-iOS/commit/86b03cb994b6f8540ea903fde712f5dab179c1ab for a bit of added consistency across apps.

File: WooCommerce/Resources/AppStoreStrings.pot
Comment (spencertransier) [reply WooCommerce/Resources/AppStoreStrings.pot]: Thank you for putting up that PR! I merged it into `release/15.1` then updated the merge branch that this PR is based on.

**Pull request merged**
